### PR TITLE
Exercise 3.1 & 3.2 completed

### PR DESCRIPTION
**Pull request by mangeshrg**
In this PR

- [x] Exercise3.1 completed
- [x] Exercise3.2 completed
- [x] Exercise3.3 completed
- [x] Exercise3.4 completed

### CODE REVIEW

File: Week-3/Exercise3.1/Exercise3.1.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    return function(...args) {
+        const key = args.toString();
+        if(cache.has(key)) {
+            return cache.get(key);
+        }
+        cache.set(key, func(...args));
+        return cache.get(key);
+    } 
+ }
+
```
Comment (chirag-udeshi) [inline Week-3/Exercise3.1/Exercise3.1.js:12]: The below add function can only accepts 2 variables hence we can only add 2 numbers. But as per Guideline 4 -> The "function should be able to take n number of argument" .

File: Week-3/Exercise3.2/Exercise3.2.js [hunk 2]
```diff
@@ -0,0 +1,11 @@
+var employee = {
+    firstName: 'John',
+    lastName: 'Doe',
+    getEmpDetails : function(dept, location) {
+        return this.firstName + ' ' + this.lastName + ' works in ' + dept + ' department at ' + location;
+    }
+ }
+
```
Comment (chirag-udeshi) [inline Week-3/Exercise3.2/Exercise3.2.js:5]: The example given for bind is correct. But as per guidelines, we have to show all implementation of bind, call & apply. Please add the latter 2 implementation.

File: Week-3/Exercise3.4/Exercise3.4.js [hunk 3]
```diff
@@ -0,0 +1,18 @@
+function createStack() {
```
Comment (chirag-udeshi) [inline Week-3/Exercise3.4/Exercise3.4.js:1]: The solution given is correct, simple & efficient. Also try implementing the solution using closures.

**Review by chirag-udeshi — APPROVED** [review]

File: Week-3/Exercise3.1/Exercise3.1.js (hunk 1 above)
Comment (mangeshrg) [reply Week-3/Exercise3.1/Exercise3.1.js:12]: @chirag-udeshi Made the changes for this.

File: Week-3/Exercise3.2/Exercise3.2.js (hunk 2 above)
Comment (mangeshrg) [reply Week-3/Exercise3.2/Exercise3.2.js:5]: @chirag-udeshi Also added the call and apply methods implementation

File: Week-3/Exercise3.1/Exercise3.1.js (hunk 1 above)
Comment (chirag-udeshi) [reply Week-3/Exercise3.1/Exercise3.1.js:12]: Looks good.

File: Week-3/Exercise3.2/Exercise3.2.js (hunk 2 above)
Comment (chirag-udeshi) [reply Week-3/Exercise3.2/Exercise3.2.js:5]: Looks good!

**Review by chirag-udeshi — APPROVED** [review]

**Pull request merged**